local_signer_example/src/client.js

```diff
@@ -0,0 +1,425 @@
+         */
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const response = (
+            await instance.post("/login", {
+                accountId: accountId != null ? accountId.toString() : null,
+            })
+        ).data;
+
+        if (response.error != null) {
+            throw new Error(response.error);
+        }
+
+        const id = AccountId.fromString(response.accountId);
+        const publicKey = PublicKey.fromString(response.publicKey);
+        const ledgerId = LedgerId.fromString(response.ledgerId);
+        const provider = new LocalProvider(
+            ledgerId,
+            response.network,
+            response.mirrorNetwork
+        );
+
+        return new LocalSigner(id, publicKey, provider);
+    }
+
+    /**
+     * @returns {Provider=}
+     */
+    getProvider() {
+        return this.provider;
+    }
+
+    /**
+     * @abstract
+     * @returns {AccountId}
+     */
+    getAccountId() {
+        return this.accountId;
+    }
+
+    /**
+     * @returns {Key}
+     */
+    getAccountKey() {
+        return this.publicKey;
+    }
+
+    /**
+     * @returns {LedgerId?}
+     */
+    getLedgerId() {
+        return this.provider == null ? null : this.provider.getLedgerId();
+    }
+
+    /**
+     * @abstract
+     * @returns {{[key: string]: (string | AccountId)}}
+     */
+    getNetwork() {
+        return this.provider == null ? {} : this.provider.getNetwork();
+    }
+
+    /**
+     * @abstract
+     * @returns {string[]}
+     */
+    getMirrorNetwork() {
+        return this.provider == null ? [] : this.provider.getMirrorNetwork();
+    }
+
+    /**
+     * @param {Uint8Array[]} messages
+     * @returns {Promise<SignerSignature[]>}
+     */
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    sign(messages) {
+        return Promise.reject(new Error("not implemented"));
+    }
+
+    /**
+     * @returns {Promise<AccountBalance>}
+     */
+    getAccountBalance() {
+        return this.sendRequest(
+            new AccountBalanceQuery().setAccountId(this.accountId)
+        );
+    }
+
+    /**
+     * @abstract
+     * @returns {Promise<AccountInfo>}
+     */
+    getAccountInfo() {
+        return this.sendRequest(
+            new AccountInfoQuery().setAccountId(this.accountId)
+        );
+    }
+
+    /**
+     * @abstract
+     * @returns {Promise<TransactionRecord[]>}
+     */
+    getAccountRecords() {
+        return this.sendRequest(
+            new AccountRecordsQuery().setAccountId(this.accountId)
+        );
+    }
+
+    /**
+     * @param {Transaction} transaction
+     * @returns {Promise<Transaction>}
+     */
+    async signTransaction(transaction) {
+        /** @type {LocalProviderResponse} */
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const response = (
```

Comment:
@sergmetelin I'm conflicted on this implementation; the issue here is that `LocalSigner.signTransaction()` communicates with the server without going through `LocalProvider`. `LocalSigner` should **only** use `this.provider` for communication, but because `Provider` doesn't expose any a `sign` method we need to implement the communication protocol in `LocalSigner.signTransaction()`. We could get around this by simply adding a `signTransaction` method to the `Provider` interface, but that sort of goes against ethers.io Speaking of which I'm not completely sure if ethers.io even supports remote signing, so maybe we shouldn't either?

cc @gregscullard in case you have any ideas for this